pkg/kademlia/kademlia.go

```diff
@@ -188,6 +188,19 @@ func (k *Kademlia) WaitForBootstrap() {
 	k.bootstrapFinished.Wait()
 }
 
+// GetPeerIdentity connects to a node and returns its peer identity
+func (k *Kademlia) GetPeerIdentity(ctx context.Context, nodeID storj.NodeID) (*identity.PeerIdentity, error) {
+	if !k.lookups.Start() {
+		return nil, context.Canceled
+	}
+	defer k.lookups.Done()
+	node, err := k.FindNode(ctx, nodeID)
+	if err != nil {
+		return nil, err
+	}
+	return k.dialer.GetPeerID(ctx, node)
```

Comment:
Why is this doing an additional dial when it could've gotten the information from `FindNode`?

Reply:
Lookup would involve changing functions in Kademlia, PeerDiscover, and Dialer.  I wasn't entirely sure if lookup connected even if the nodeid was already in the routing table.  This was way easier. :)